File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/outbound/events/EventPublisher.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.events;

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.*;

public interface EventPublisher {

    void onDoctorCreated(DoctorCreated event);

    void onPatientCreated(PatientCreated event);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/outbound/events/EventPublisher.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.events

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.events.*

interface EventPublisher {

    fun onDoctorCreated(event: DoctorCreated)

    fun onPatientCreated(event: PatientCreated)
}
